### PR TITLE
Add Lock database on Screen Sleep

### PR DESCRIPTION
**Pull request by georgesnow**
@mstarke added the functionality where the database can lock when the **screen** goes to sleep. My Mac never sleeps nor does it log out (or at least rarely)....haha, but I use the lock screen which after 30 seconds puts my displays to sleep. anyway bugged me that my database wasn't locking. I had few instance where I got couple of bad merges of databases when I went to sue other machines. And this would fix that.

also the **_constraints  on the preferences window need to be set_**. I couldn't figure out how to do it...sorry

### CODE REVIEW

File: MacPass/MPLockDaemon.m [hunk 1]
```diff
@@ -95,6 +97,18 @@ - (void)setLockOnSleep:(BOOL)lockOnSleep {
...
 }
 
+- (void)setLockOnScreenSleep:(BOOL)lockOnScreenSleep {
+  if(_lockOnScreenSleep != lockOnScreenSleep) {
+    _lockOnScreenSleep = lockOnScreenSleep;
+    if(_lockOnScreenSleep) {
+      [NSWorkspace.sharedWorkspace.notificationCenter addObserver:self selector:@selector(_willScreenSleepNotification:) name:NSWorkspaceScreensDidSleepNotification object:nil];
+    }
+    else {
+      [NSWorkspace.sharedWorkspace.notificationCenter removeObserver:self name:NSWorkspaceWillSleepNotification object:nil];
```
Comment (mstarke) [inline MacPass/MPLockDaemon.m]: You registered for ```NSWorkspaceScreensDidSleepNotification``` but unregister for ```NSWorkspaceWillSleepNotification```

Comment (georgesnow) [reply MacPass/MPLockDaemon.m]: Bad paste. I will fix that

**Comment by georgesnow** [comment]
@mstarke fixed.

**Comment by georgesnow** [comment]
@mstarke hey just wondering - other than the constraints on the preference page (which I might take another stab at). is there anything keeping this pull request from being merged?

**Comment by georgesnow** [comment]
I ask cause I would like to get in the next continuous build since that tends to be my “everyday” version of Macpass

**Comment by georgesnow** [comment]
@mstarke so this should be **all good including constraints** to merge. I added the proper constraints as well as fixing a label error I hadn't noticed. can we go ahead now and merge this pull request?

**Comment by mstarke** [comment]
I've merged but I need to clean up the fonts. I have no clue why you changed them to something non-default

**Comment by georgesnow** [comment]
@mstarke _**I did not intend to.**_ I  literally just cloned the repo and added the code. Then copied one of the other check boxes to make sure I didn't change any fonts etc...sorry

**Comment by mstarke** [comment]
I thought that's what you did but I wasn't able to get the same results. There where multiple inconsistencies in the font settings which I suspect are there because of migrating the codebase along the way.  Some other issues with the outlet not being connected properly existed as well which are all fixed now so the feature works in master. Continuous will get updated as well shortly.

**Comment by georgesnow** [comment]
@mstarke awesome thanks! i must have merge improperly from the main fork or my Xcode settings. sorry I was trying to fix the pieces I had left undone. this experience makes me nervous when I attempt to merge Touch ID code that there are going to be a ton of inconsistencies. I will need to resolve.

**Comment by mstarke** [comment]
Welcome to the world of app developers :). You might get an idea why I tend to not rush features but instead sit many out a bit longer to get things right (or leave them out altogether)

**Pull request merged**